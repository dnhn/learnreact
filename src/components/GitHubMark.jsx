import React from 'react';

import './GitHubMark.scss';

import mark from '../static/images/github-mark.svg';
import markLight from '../static/images/github-mark-light.svg';

export default ({ toggle }) => (
  <a
    href="https://github.com/nhantdn/learnreact"
    target="_blank"
    rel="noopener noreferrer"
    className="logo"
  >
    <img src={toggle ? mark : markLight} alt="" />
  </a>
);
