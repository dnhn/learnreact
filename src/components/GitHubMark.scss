.logo {
  z-index: 1;
  position: absolute;
  overflow: hidden;
  right: 0;
  top: 40px;
  line-height: 0;

  &:hover {
    img {
      transform: rotate(-135deg) translateY(5px);
    }
  }

  img {
    width: 48px;
    height: 48px;
    transform: rotate(-135deg) translateY(25px);
    transition: transform .25s ease-in-out;
  }

  @media screen and (max-width: 640px) {
    position: fixed;

    img {
      transform: rotate(-135deg) translateY(5px);
    }
  }
}
