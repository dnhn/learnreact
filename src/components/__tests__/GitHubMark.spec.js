import React from 'react';
import { render } from '@testing-library/react';
import GitHubMark from '../GitHubMark';

const rootUrl = 'http://localhost/';

it('renders correctly', () => {
  const { container } = render(<GitHubMark />);
  expect(container).toMatchSnapshot();
  expect(container.getElementsByTagName('img')[0].src)
    .toBe(`${rootUrl}github-mark-light.svg`);
});

it('toggles style', () => {
  const { container } = render(<GitHubMark toggle />);
  expect(container.getElementsByTagName('img')[0].src)
    .toBe(`${rootUrl}github-mark.svg`);
});
