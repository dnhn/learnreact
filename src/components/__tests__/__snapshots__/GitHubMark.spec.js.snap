// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GitHubMark /> renders correctly 1`] = `
<div>
  <a
    class="logo"
    href="https://github.com/nhantdn/learnreact"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Hosted by GitHub"
      src="github-mark-light.svg"
    />
  </a>
</div>
`;
